ifrs17/Utils/Queries.ipynb

```diff
@@ -399,6 +399,67 @@
             "metadata": {},
             "execution_count": 0,
             "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Data Variables"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> LoadPartitionedDataAsync<T,P>(this IDataSource querySource, Guid partition)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    await querySource.Partition.SetAsync<P>(partition);",
+                "\n    return (await querySource.Query<T>().ToArrayAsync());",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> RelaxedQueryPartitionedDataAsync<T,P>(this IWorkspace workspace, IDataSource dataSource, Guid targetPartition, Guid defaultPartition, string format)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    var isRelaxed = (format != ImportFormats.Cashflow && typeof(T).Name == nameof(IfrsVariable)) || ",
+                "\n                    (format == ImportFormats.Cashflow && typeof(T).Name == nameof(RawVariable));",
+                "\n",
+                "\n    var variablesFromWorkspace = await workspace.LoadPartitionedDataAsync<T,P>(targetPartition);",
+                "\n    if(targetPartition == defaultPartition || !isRelaxed || variablesFromWorkspace.Any()) return variablesFromWorkspace; ",
```

Comment:
I find this bool statement confusing: 
in the case the bools in this if are false,false,true => is sens, is relaxed, there are variable in the workspace, then we return the variables from the workspace. In fact the result is not relaxed. So why was the isRelaxed variable true?

Reply:
this is now a lot improved! Please check